### PR TITLE
[CSL-1139] Return last main block on GetHeaders [] request

### PR DESCRIPTION
**Pull request by volhovM**

### CODE REVIEW

File: src/Pos/Block/Network/Announce.hs [hunk 1]
```diff
@@ -87,13 +88,21 @@ handleHeadersCommunication conv _ = do
...
                 (c1:cxs, _)   -> getHeadersFromManyTo (c1:|cxs) mghTo
             either onNoHeaders handleSuccess headers
   where
+    -- retrieves last main block if there's any
+    getLastMainBlock :: m (BlockHeader ssc)
+    getLastMainBlock = do
+        (tip :: Block ssc) <- DB.getTipBlock @ssc
+        let tipHash = tip ^. blockHeader
+        case tip of
+            Left _  -> fromMaybe tipHash <$> DB.getBlockHeader (tip ^. prevBlockL)
```
Comment (Martoon-00) [inline src/Pos/Block/Network/Announce.hs]: If this is first genesis block, you take it?

File: src/Pos/Block/Network/Announce.hs [hunk 2]
```diff
@@ -87,13 +88,21 @@ handleHeadersCommunication conv _ = do
...
                     maybeToRight "getBlockHeader returned Nothing" . fmap one <$>
                     DB.getBlockHeader @ssc h
                 (c1:cxs, _)   -> getHeadersFromManyTo (c1:|cxs) mghTo
             either onNoHeaders handleSuccess headers
   where
+    -- retrieves last main block if there's any
+    getLastMainBlock :: m (BlockHeader ssc)
+    getLastMainBlock = do
+        (tip :: Block ssc) <- DB.getTipBlock @ssc
+        let tipHash = tip ^. blockHeader
```
Comment (gromakovsky) [inline src/Pos/Block/Network/Announce.hs]: Why is it called `tipHash` if it is a header? Can you rename it? It's confusing. Especially without explicit type.

File: src/Pos/Block/Network/Announce.hs [hunk 3]
```diff
@@ -87,13 +88,21 @@ handleHeadersCommunication conv _ = do
...
             headers <- case (mghFrom,mghTo) of
-                ([], Nothing) -> Right . one <$> DB.getTipBlockHeader @ssc
+                ([], Nothing) -> Right . one <$> getLastMainBlock
                 ([], Just h)  ->
                     maybeToRight "getBlockHeader returned Nothing" . fmap one <$>
                     DB.getBlockHeader @ssc h
                 (c1:cxs, _)   -> getHeadersFromManyTo (c1:|cxs) mghTo
             either onNoHeaders handleSuccess headers
   where
+    -- retrieves last main block if there's any
```
Comment (gromakovsky) [inline src/Pos/Block/Network/Announce.hs]: Not block but header.

File: src/Pos/Block/Network/Announce.hs (hunk 1 above)
Comment (volhovM) [inline src/Pos/Block/Network/Announce.hs]: It's not the first genesis block in a normal environment, first. Second, previous implementation would do that anyway.

File: src/Pos/Block/Network/Announce.hs (hunk 2 above)
Comment (volhovM) [inline src/Pos/Block/Network/Announce.hs]: Yes, something went wrong with auto-renaming i guess.

**Pull request merged**